src/one_dragon/envs/git_service.py

```
@@ -860,6 +880,14 @@ def _fetch_remote(
                 self.env_config.last_repository_url = used_repository.url
             except Exception:
                 log.warning('记录上次成功代码源失败', exc_info=True)
+            # 使用代理时记录本次成功的线路，下次优先尝试
+            if self.env_config.is_gh_proxy and used_repository.use_proxy:
+                used_proxy_url = self._extract_gh_proxy(used_url, used_repository)
+                if used_proxy_url is not None:
+                    try:
+                        self.env_config.gh_proxy_url = used_proxy_url
+                    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)